coordination/src/main/java/tech/ydb/coordination/CoordinationClient.java

```diff
@@ -0,0 +1,68 @@
+package tech.ydb.coordination;
+
+import java.util.concurrent.CompletableFuture;
+
+import tech.ydb.coordination.impl.CoordinationClientImpl;
+import tech.ydb.coordination.rpc.CoordinationRpc;
+import tech.ydb.coordination.session.CoordinationSession;
+import tech.ydb.coordination.session.LeaderElectionSession;
+import tech.ydb.coordination.settings.CoordinationNodeSettings;
+import tech.ydb.coordination.settings.DescribeCoordinationNodeSettings;
+import tech.ydb.coordination.settings.DropCoordinationNodeSettings;
+import tech.ydb.coordination.settings.SessionSettings;
+import tech.ydb.core.Status;
+
+/**
+ * @author Kirill Kurdyukov
+ */
+public interface CoordinationClient {
+
+    static CoordinationClient newClient(CoordinationRpc coordinationRpc) {
+        return new CoordinationClientImpl(coordinationRpc);
+    }
+
+    /**
+     * Bidirectional stream used to establish a session with a coordination node
+     * <p>
+     * Relevant APIs for managing semaphores, distributed locking, creating or
+     * restoring a previously established session are described using nested
+     * messages in SessionRequest and SessionResponse. Session is established
+     * with a specific coordination node (previously created using CreateNode
+     * below) and semaphores are local to that coordination node.
+     */
+    CoordinationSession createSession();
+
+    CompletableFuture<LeaderElectionSession> createLeaderElectionSession(SessionSettings settings);
```

Comment:
We must not mix low level api and hi level api in one place. We have CoordationClient as the base implementation of the coordination service, and we can have a few helpers as LeaderElection, which use this service, but they aren't a part of the coordination service.